packages/entities/entities-routes/src/components/RouteFormRulesComposer.vue

```diff
@@ -247,19 +321,55 @@ const payload = computed<TypedRouteRulesPayload | undefined>(() => {
   }
 })
 
+const onProtocolUpdate = (protocol: string | number | null) => {
+  if (protocol) {
+    emit('update:protocols', protocol as keyof typeof PROTOCOLS_TO_ROUTE_RULES)
+  }
+}
+
+watch([() => props.recordFlavor, () => props.routeFlavors], ([newRecordFlavor, newRouteFlavors]) => {
+  if (newRecordFlavor) {
+    configFlavor.value = newRecordFlavor
+    return
+  }
+  if (newRouteFlavors.traditional) {
```

Comment:
```suggestion
 if (newRouteFlavors.expressions) {
    configFlavor.value = RouteFlavor.EXPRESSIONS
  } else {
    configFlavor.value = RouteFlavor.TRADITIONAL
  }
```

Reply:
If the value of `newRouteFlavors` is `{ traditional: true,  expressions: true }`, we still want `configFlavor.value` to be `RouteFlavor.TRADITIONAL`. But in the suggested change, `configFlavor.value` will be `RouteFlavor.EXPRESSIONS`, which is not intended.

Reply:
Got it. I'm missing that.

Reply:
Wouldn't it be easier to understand if we also added `newRouteFlavors.expressions` in the first branch?

Reply:
I simplified this block in https://github.com/Kong/public-ui-components/pull/2111/commits/84c943bf0c308077cfb66eb3e88c21d3ee3ddadb.